### DISCUSSION TITLE
Set Slot width

### DISCUSSION THREAD

**Discussion by AlexHedley** [microsoft/fluentui-blazor#3079] [Q&A]
Is there a way to set the size of the slot width on a `FluentTextField`?

- https://www.fluentui-blazor.net/TextField

**Comment by vnbaaij** [comment]
We don't supply anything for that out of the box but you should be able to change it through CSS

**Reply by AlexHedley** [reply]
Would that just be with a `style="width: 25px;"` on the control that you have the `slot` against it?

For example on the `FluentIcon` below.

```html
<FluentTextField @bind-Value=value1>
    <FluentIcon Value="@(new Icons.Regular.Size16.Globe())" Slot="start" Color="Color.Neutral" />
</FluentTextField>
```

**Reply by vnbaaij** [reply]
Probably yes.Don't know (not at pc). Have you tried?

**Reply by AlexHedley** [reply]
Me neither, I'll give it a go later and report back.
Thanks for the suggestion.

**Reply by AlexHedley** [reply]
I tried it on the control, but it didn't change the slot size.

```html
<FluentTextField TextFieldType="TextFieldType.Number" Value="123" Disabled="true">
    <FluentLabel Slot="start" Style="width=200px;" Typo="Typography.Body">Message Id</FluentLabel>
</FluentTextField>

<br />

<FluentTextField TextFieldType="TextFieldType.Number" Value="234" Disabled="true">
    <FluentLabel Slot="start" style="width=200px;" Typo="Typography.Body">Message Id</FluentLabel>
</FluentTextField>

<br />

<FluentTextField TextFieldType="TextFieldType.Number" Value="345" Disabled="true">
    <FluentLabel Slot="start" Typo="Typography.Body">Message Id</FluentLabel>
</FluentTextField>
```

Output

```html
<fluent-text-field type="number" id="f6a3c101f" value="123" current-value="123" disabled="" appearance="outline" class="outline disabled" _bl_134="">
    <!--!-->
    <p slot="start" typo="body" class="fluent-typography" style="width=200px;" b-1nnnfjehkp="">Message Id</p>
</fluent-text-field>
```

---

Once the page had loaded I manually updated the `#shadow-root`.

```html
<span part="start" class="start" style="width: 150px;"><slot name="start"></slot></span>
```
![image](https://github.com/user-attachments/assets/1d86100b-d16c-4062-ad8b-a9fa363dd252)

---

I tried adding css to the `.razor.css` for the page but couldn't see it the `<link href="[PROJECTNAME].styles.css" rel="stylesheet" />` but that's a me problem I'll look into as well.

```css
.start {
    width: 150px;
}
```

I'll keep trying other ideas.

**Reply by vnbaaij** [reply]
You'll need to use a specific syntax to style what is in the shadow dom. Something like:
```
::deep fluent-text-field::part(start) {...}
```

**Reply by AlexHedley** [reply]
Oh something like https://github.com/microsoft/fluentui-blazor/blob/b189dc3d6b1dfe9cb7c316f59276ab5b7625f76c/src/Core/Components/List/FluentAutocomplete.razor.css#L23
I'll give that a try.
